modules/core/shared/src/main/scala/lucuma/core/math/SignalToNoise.scala

```diff
@@ -80,6 +92,14 @@ object SignalToNoise {
   val FromBigDecimalRounding: Format[BigDecimal, SignalToNoise] =
     Format(bd => fromMilliDecimal((bd * 1000).setScale(0, BigDecimal.RoundingMode.HALF_UP)), _.toBigDecimal)
 
+  /**
+   * Creates a `SignalToNoise` value assuming that the given PosBigDecimal is in
+   * range [Min, Max].  Rounds finer scale values to milli-sn.
+   *
+   * @group Optics
+   */
+  val FromPosBigDecimalRounding: Format[PosBigDecimal, SignalToNoise] =
```

Comment:
May I suggest using a `ValidSplitEpi` here? It combines the functionality of `Format` and `SplitEpi`: the valid inputs can be normalized.

Probably `FromBigDecimalRounding` above too.

I suspect that going this way we won't need custom generators in the tests.